### ISSUE TITLE
Switch to vi mode in open shell

### ISSUE THREAD

**Issue by robertocarta** [ipython/ipython#11329]
Hi, I use ipython for my django shell in shared machines. With readline you can "set editing_mode=vi" or something along those lines, is there a way to do it with the current ipython with prompt_toolkit?

**Comment by Carreau** [comment]
Yes, you can use `--TerminalInteractiveShell.editing_mode=vi` or set `c.TerminalInteractiveShell.editing_mode='vi'` in your configuration option. 

It is not perfect, but ressemble VI.

You can switch dynamically with something like:

```
In [1]: ip= get_ipython()
In [2]: ip.editing_mode
Out[2]: 'emacs'
In [3]: ip.editing_mode='vi'
```

There is no mode indicator, I'm sure there is an issue somewhere about that. If you figure it out feel free to send PR, or at least pointer. And/Or send updates to the docs. 

In general `ipython --help-all` and search through the output can be useful to figure out these can of things.

**Comment by robertocarta** [comment]
I knew about c.TerminalInteractiveShell.editing_mode, but I needed to do it dynamically on an open ipython. I tried your method, but it doesn't seem to work. Maybe your config includes something that auto reloads settings on change?
(but, I'll definitely work on a PR, having an easy switch cold be useful to many)

**Comment by robertocarta** [comment]
Ok the missing bit was ip.init_prompt_toolkit_cli()

**Comment by bitan2988** [comment]
Yes, Use these following commands --TerminalInteractiveShell.editing_mode=vi or set c.TerminalInteractiveShell.editing_mode='vi' in your configuration option.



You can switch dynamically with the help of these command:

In [1]: ip= get_ipython()
In [2]: ip.editing_mode
Out[2]: 'emacs'
In [3]: ip.editing_mode='vi'

**Comment by Carreau** [comment]
See #11390 which fixes the dynamic switching with `%config TerminalInteractiveShell.editing_mode='vi'`

**Comment by Carreau** [comment]
closed by #11390
